app/utils/dates.ts

```diff
@@ -34,13 +34,32 @@ export function weekNumberToDate({
   return result;
 }
 
+/**
+ * Retrieve a valid date. If invalid, get a new Date object.
+ *
+ * Returns:
+ * - The date object if valid
+ * - Otherwise, returns `new Date()`
+ */
+export function getValidNewDateIfInvalid(date: Date) {
+  if (isNaN(date.getTime())) {
+    console.error("Invalid date, returning a new Date object");
```

Comment:
Agreed, I'll refactor this date validation function

Reply:
It has been refactored, though I left TODO comments for https://github.com/Sendouc/sendou.ink/issues/1082

I left the console.warn() calls for invalid dates, but commented out any thrown errors because of an invalid date